doc/tutorial.md

```diff
@@ -0,0 +1,607 @@
+
+we will create a folder called _platform_launch_ that contains the following files:
+
+- _main.c_
+- _Makefile.am_
+- _configure.am_
+##setup the build system
+starting from the build part:
+_configure.am_:
+```
+m4_define([package_version_major],[0])
+m4_define([package_version_minor],[0])
+m4_define([package_version_micro],[0])
+
+AC_INIT([platform_launch], [package_version_major.package_version_minor.package_version_micro])
+AM_INIT_AUTOMAKE([foreign -Wall])
+
+# compilers
+AC_PROG_CC
+
+PKG_PROG_PKG_CONFIG
+PKG_INSTALLDIR
+
+AC_CONFIG_HEADERS([config.h])
+AC_CONFIG_MACRO_DIR([m4])
+
+# Check if the `install` program is present
+AC_PROG_INSTALL
+
+m4_ifdef([AM_PROG_AR], [AM_PROG_AR])
+LT_INIT(disable-static)
+
+PKG_CHECK_MODULES(UBX, ubx0 >= 0.6.0)
+
+PKG_CHECK_VAR([UBX_MODDIR], [ubx0], [UBX_MODDIR])
+  AC_MSG_CHECKING([ubx module directory])
+  AS_IF([test "x$UBX_MODDIR" = "x"], [
+  AC_MSG_FAILURE([Unable to identify ubx module path.])
+])
+AC_MSG_RESULT([$UBX_MODDIR])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
+```
+
+_Makefile.am_
+```cmake
+ubxmoddir = ${UBX_MODDIR}
+
+ACLOCAL_AMFLAGS= -I m4
+ubxmod_PROGRAMS = platform_main
+platform_main_SOURCES = $(top_srcdir)/libubx/ubx.h  main.c
+platform_main_CFLAGS = -I${top_srcdir}/libubx  @UBX_CFLAGS@
+platform_main_LDFLAGS = -module -avoid-version -shared -export-dynamic  @UBX_LIBS@ -ldl
+```
+
+Here we specifiy that the name of the executable is _platform_main_
+It might be possible that, if some custom types are used in the configuration, but are not installed, they mist be added to the _CFLAGS_:
+
+```cmake
+platform_main_CFLAGS = -I${top_srcdir}/libubx -I path/to/other/headers  @UBX_CFLAGS@
+```
+
+In order to compile, we will use the same commands as before (we do not need to install).
+```bash
+autoreconf --install
+./configure
+make
+```
+
+## The program
+The main follows the same structure of the .usc file.
+### Libraries
+It start with some include (struct that are needed in configuration) and loading the libraries
+```c
+#include <ubx.h>
+
+#define WEBIF_PORT "8810"
+#define DOUBLE_STR "double"
+#include "ptrig_config.h"
+#include "ptrig_period.h"
+int main()
+{
+  int len, ret=EXIT_FAILURE;
+  ubx_node_info_t ni;
+  ubx_block_t *plat1, *control1, *logger1, *ptrig1, *webif, *fifo_vel, *fifo_pos;
+  ubx_data_t *d;
+
+  /* initalize the node */
+  ubx_node_init(&ni, "platform_and_control");
+const char *modules[8];
+modules[0]= "/usr/local/lib/ubx/0.6/stdtypes.so";
+modules[1]= "/usr/local/lib/ubx/0.6/stattypes.so";
+modules[2]= "/usr/local/lib/ubx/0.6/ptrig.so";
+modules[3]= "/usr/local/lib/ubx/0.6/platform_2dof.so";
+modules[4]= "/usr/local/lib/ubx/0.6/platform_2dof_control.so";
+modules[5]= "/usr/local/lib/ubx/0.6/webif.so";
+modules[6]= "/usr/local/lib/ubx/0.6/logger.so";
+modules[7]= "/usr/local/lib/ubx/0.6/lfds_cyclic.so";
+  /* load modules */
+for (int i=0; i<8;i++)
+  if(ubx_module_load(&ni, modules[i]) != 0){
+      printf("fail to load %s",modules[i]);
```

Comment:
Out of the block context you can't use the `ubx_err` functions, but you can still use the `ubx_log` function as described here:
https://github.com/kmarkus/microblx/blob/dev/doc/manual.md#using-real-time-logging